pkg/rollout/traffic.go

```diff
@@ -0,0 +1,184 @@
+package rollout
+
+import (
+	"time"
+
+	"github.com/GoogleCloudPlatform/cloud-run-release-operator/pkg/health"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/api/run/v1"
+)
+
+// trafficBasedOnDiagnosis updates a service's traffic configuration
+// based on the diagnosis.
+// If traffic should not changed, nil is returned.
+func (r *Rollout) trafficBasedOnDiagnosis(svc *run.Service, diagnosis health.DiagnosisResult, stable, candidate string) ([]*run.TrafficTarget, error) {
```

Comment:
determineTraffic["",Targets,Rules] any of these are better names than this one IMO